tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -1916,6 +1916,62 @@ def visit_module(self, node):
                 logger.debug("Pylint custom checker failed to check if model is aliased.")
                 pass
 
+class CheckExceptionLogging(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "check-exception-logging"
+    priority = -1
+    msgs = {
+        "C4751": (
+            "Do not allow logging of Exception, as it can reveal personally identifiable information. See Details: "
+            "https://azure.github.io/azure-sdk/python_implementation.html#python-logging-sensitive-info",
+            "exception-logging",
+            "Do not log exceptions with anything other than DEBUG.",
+        ),
+    }
+    options = (
+        (
+            "ignore-exception-logging",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Do not log exceptions with anything other than DEBUG.",
+            },
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(CheckExceptionLogging, self).__init__(linter)
+
+    def visit_tryexcept(self, node):
+        logger_names = ["_logger", "logger"]
+        uses_exception = False
+        if isinstance(node.handlers[0], astroid.ExceptHandler) and node.handlers[0].type:
```

Comment:
Why does this have to check that the handler has a type on it? Are there cases where it doesnt?

Reply:
Also is it always the first handler?

Reply:
I will investigate!